### PR TITLE
Fix Bug 1249610, add new infobar to bedrock

### PR DESCRIPTION
**Pull request by schalkneethling**
Couple of notes:

The code for the new infobar lives at:
https://github.com/mozilla/mozilla-infobar/

I am mentioning that, because the code has thus already been reviewed. With that said though, I have made some changes based on bugs and edge cases I ran into whilst integrating it into Bedrock so, looking over the code during this pull request works for me.

Next thing, I have spot checked a bunch of pages (I will be sharing a Google Sheet with the details).

I tested the translation bar using en-US, de and fr. I switched locales using about:config in Fx, as I cannot find an addon that actually works.

I did not test this in any other browsers. The only bar that should appear in these will be the Translation Bar. I did test that the update bar does not show.

As with Fx, I cannot find a language switcher that works as advertised. I did attempt to adjust the language setting using Language and input settings in Chrome but, this does not have the desired effect. Any suggestions here are welcome.

@alexgibson @craigcook @jpetto @kyoshino

### CODE REVIEW

File: media/js/infobar/infobar.js [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+        $acceptButton: {},
+        $cancelButton: {},
+        // info bar(s) the user wish to use
+        bars: [],
+        // The current instance of an InfoBar
+        bar: {},
+        opened: false,
+        json: {},
+        pageLang: '',
+        // list of available languags from either a list of hreflang links
```
Comment (flodolo) [inline media/js/infobar/infobar.js]: typo (languages)

File: media/js/infobar/infobar.js [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+
+    /**
+     * Returns selected language's BiDi layout.
+     * False = left-to-right layout
+     * True = right-to-left layout
+     * @param {sting} lang - A language string such as en-US
+     */
+    InfoBar.prototype.getLanguageBidi = function(lang) {
+        // Languages using BiDi (right-to-left) layout
+        var LANGUAGES_BIDI = ['he', 'ar', 'fa', 'ur'];
```
Comment (flodolo) [inline media/js/infobar/infobar.js]: I would order this...

File: media/js/infobar/infobar.js [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+     * @param {string} lang - The language string to be normalized
+     */
+    InfoBar.prototype.normalize = function (lang) {
+        return lang.replace(/^(\w+)(?:-(\w+))?$/, function (m, p1, p2) {
+            return p1.toLowerCase() + ((p2) ? '-' + p2.toUpperCase() : '');
+        });
+    };
+
+    /**
+     * Gets and sets the list of available languags from either a list of hreflang links
```
Comment (flodolo) [inline media/js/infobar/infobar.js]: Same typo

File: media/js/infobar/infobar.js [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+            });
+            return availableLangs;
+        } else {
+            // If those lists cannot be found, there is nothing to do
+            return false;
+        }
+    };
+
+    /**
+     * Determines whether the users accept language(s) match the current
```
Comment (flodolo) [inline media/js/infobar/infobar.js]: user's accept language list matches

File: media/js/infobar/infobar.js [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+                offeredLang = userLang;
+            }
+        });
+
+        // If there is no offered language return false, else true
+        return !offeredLang ? false : offeredLang;
+    };
+
+    /**
+     * Get's the list of the user's acceptLanguages as appropriate for the
```
Comment (flodolo) [inline media/js/infobar/infobar.js]: Get

File: media/js/infobar/infobar.js [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+     */
+    InfoBar.prototype.isLikeFirefox = function(userAgent) {
+        var ua = userAgent || navigator.userAgent;
+        return (/Iceweasel/i).test(ua) || (/IceCat/i).test(ua) ||
+            (/SeaMonkey/i).test(ua) || (/Camino/i).test(ua) ||
+            (/like Firefox/i).test(ua);
+    };
+
+    /**
+     * Rturns true if this is a version of Firefox mobile excluding Firefox for iOS
```
Comment (flodolo) [inline media/js/infobar/infobar.js]: Returns

**Comment by schalkneethling** [comment]
I have made an update to the code. It now determined the pageLang match as well as the offered language using only the first entry in the acceptLanguages array.

This means that, in Chrome, going to about:settings -> Language and input settings and adjusting your preferred language there, now works as expected.

**Comment by schalkneethling** [comment]
I am working on adding more, porting, unit tests, and have found the following test to determine whether the user's and the page language match:

```
it('should return false if the user\'s language is the same as the page\'s language',
...
expect(setup(['ja', 'pt-PT', 'el', 'fr', 'en'], 'el')).toBeFalsy();
```

The above suggests that, if the page language is `el`, we have a match, because it is one of the entries in the accept languages array. This means, we should indeed be looping over all of the accept languages in the `userLangPageLangMatch` function for example.

Agree, disagree ;) Let me know, thx

**Comment by jpetto** [comment]
@schalkneethling - I'm going to bring this up in our next sprint planning meeting. Is the work here primarily finished? Looks like we're waiting on UX approval in the bug. If the code is essentially complete (aside from a rebase), I'll suggest we get final approval on moving forward. If not, I'll get it properly prioritized (so it can stop staring us in the face at every PR triage).

Thanks!

**Comment by schalkneethling** [comment]
@jpetto I believe the code is there. The main things that was still needed was more/better test coverage, manual testing of the integration into the site, and then there was some UX work that needed to be done.

**Comment by jpetto** [comment]
Hmm, ok. Aside from the UX work, could you estimate about how much dev time remains? Feel free to be broad - estimating in half day increments would be just fine.

**Comment by schalkneethling** [comment]
@jpetto I would say 3 days should be sufficient.

**Comment by craigcook** [comment]
This pull request is almost a year old at this point and the future of the infobar is still uncertain. There's some current strategy forming around a revamped cross-site navbar that may take its place, as well as the new branding and design system coming soon that could render this even further out-of-date, so I don't think this PR can ever be merged. We're going to close it now. 

If you listen closely you'll hear the sound of a single bugle softly playing Taps in the distance 🎺

**Pull request closed without merging**